proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -8847,24 +8847,34 @@ lemma bound_head_time_buffer:
   done
 
 lemma head_time_buffer_true_imp_buffer:
-  "\<lbrakk>the (head_time_buffer usage s); pred_map \<top> (sc_refill_cfgs_of s) (cur_sc s)\<rbrakk>
-   \<Longrightarrow> pred_map (\<lambda>cfg. max_time - r_time (hd ((scrc_refills cfg))) \<ge> 5 * MAX_PERIOD)
-                (sc_refill_cfgs_of s) (cur_sc s)"
-  apply (frule bound_head_time_buffer[where usage=usage])
-  apply (clarsimp simp: head_time_buffer_def obind_def read_sched_context_def ogets_def
-                        vs_all_heap_simps
-                 split: kernel_object.splits)
+  "pred_map \<top> (sc_refill_cfgs_of s) (cur_sc s)
```

Comment:
Since a fragment of an SC cannot exist in the kheap without an whole SC existing there, I think this should be phrased in terms of `scs_of`, probably simply something like `bound (scs_of s) (cur_sc s)`. It could be `sc_at` even.